sklearn/decomposition/_nmf.py

```diff
@@ -1183,6 +1201,19 @@ def _check_params(self, X):
         self._beta_loss = _beta_loss_to_float(self.beta_loss)
 
     def _check_w_h(self, X, W, H, update_H):
+        # n_components
+        if self._n_components == "auto":
+            if W is not None:
+                if H is not None:
+                    self._n_components = H.shape[0]
```

Comment:
Here I would raise an error if shapes don't match (H.shape[0] and W.shape[1])

Reply:
added in [261e5404ac63961856c628ef82c34f4979eea018](https://github.com/scikit-learn/scikit-learn/pull/26634/commits/261e5404ac63961856c628ef82c34f4979eea018)